src/main/haskell/kore/src/Kore/Builtin/List.hs

```diff
@@ -196,16 +210,41 @@ evalConcat :: Builtin.Function
 evalConcat =
     Builtin.functionEvaluator evalConcat0
   where
+    ctx = "LIST.concat"
+    evalConcat0
+        :: MetadataTools Object StepperAttributes
+        -> PureMLPatternSimplifier Object variable
+        -> Kore.Sort Object
+        -> [Kore.PureMLPattern Object variable]
+        -> Simplifier (AttemptedFunction Object variable)
     evalConcat0 _ _ resultSort = \arguments ->
         getAttemptedFunction
         (do
             let (_list1, _list2) =
                     case arguments of
                         [_list1, _list2] -> (_list1, _list2)
-                        _ -> Builtin.wrongArity "LIST.concat"
-            _list1 <- expectBuiltinDomainList "LIST.concat" _list1
-            _list2 <- expectBuiltinDomainList "LIST.concat" _list2
-            returnList resultSort (_list1 <> _list2)
+                        _ -> Builtin.wrongArity ctx
+                leftIdentity = do
+                    _list1 <- expectBuiltinDomainList ctx _list1
```

Comment:
In this and `Set`, `Map`, etc.: is it a problem that we run `expectBuiltinDomainList` up to twice more than we need to in case it goes to the `bothConcrete` case?

Reply:
No, `expectBuiltinDomainList` is just a wrapper to check for certain types of error, it doesn't actually do anything. It might be a little inefficient, but not much.